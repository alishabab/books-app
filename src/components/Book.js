/* eslint-disable no-console */
/* eslint-disable react/prop-types */
import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';

const Book = () => {
  const { id } = useParams();
  const books = useSelector(state => state.books);
  const book = books.find(book => book.id === id);
  return (
    <div>
      <img src={book.volumeInfo.imageLinks.thumbnail} alt={book.volumeInfo.title} />
      <p>{book.volumeInfo.title}</p>
    </div>
  );
};

export default Book;
