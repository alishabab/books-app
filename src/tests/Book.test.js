import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Book from '../components/Book';

const mockStore = configureStore([]);

describe('Book React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      books: [
        {
          kind: 'books#volume',
          id: 'kIYSAAAAIAAJ',
          title: 'Thesavrvs Lingvae Latinae Compendiarivs Or, A Compendious Dictionary of the Latin Tongue',
          smallThumbnail: 'http://books.google.com/books/content?id=kIYSAAAAIAAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
          thumbnail: 'http://books.google.com/books/content?id=kIYSAAAAIAAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
          authors: [
            'Robert Ainsworth',
          ],
        },
      ],
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <Router>
          <Book />
        </Router>
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should not dispatch action if books length is greather than 0', () => {
    expect(store.dispatch).toHaveBeenCalledTimes(0);
  });
});
