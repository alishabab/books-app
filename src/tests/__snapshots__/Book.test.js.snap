// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book React-Redux Component should render with given state from Redux store 1`] = `<div />`;
